### PR TITLE
CGP [0021]

### PR DESCRIPTION
**Pull request by anna-carroll**

### CODE REVIEW

File: CGPs/0020.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+- Date: 2021-01-29
+- Author(s): @anna-carroll
+- Status: DRAFT
+- Governance Proposal ID #: [if submitted]
+- Date Executed: [if executed]
+- Discussions To: [TODO: create and add discussion link]
+
+## Overview
+
+This proposal will extend the expiration of a requested attestation from 1 HOUR -> 1 DAY, improving upon CGP 0015 which previously attempted to do so.
```
Comment (nambrot) [inline CGPs/0020.md:12]: I would maybe add a word like `effectively` or `indirectly` in here

File: CGPs/0020.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+The initial CGP 0015 attempted to extend the expiration time of attestations from 1 hour to 24 hours by changing attestationExpiryBlocks in the Attestation contract. However, the Random contract only retains randomness for a period of blocks specified by randomnessBlockRetentionWindow. Though the Attestation contract was updated to be able to query randomness between 1 and 24 hours, the Random contract remains unable to serve those requests. Therefore, the expiration time for attestations remains functionally 1 hour.
+
+See CGP 0015 for further context on the motivation for extending the expiration of requested attestations.
```
Comment (nambrot) [inline CGPs/0020.md:16]: It's a good idea to link to CGP15, but maybe one short sentence of CGP15's motivation would make sense here as well

**Review by nambrot — APPROVED** [review]

File: CGPs/0020.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# CGP [0020]: Update randomnessBlockRetentionWindow to Extend Attestation Expiration Duration
+
+- Date: 2021-01-29
+- Author(s): @anna-carroll
+- Status: DRAFT
+- Governance Proposal ID #: [if submitted]
+- Date Executed: [if executed]
+- Discussions To: [TODO: create and add discussion link]
```
Comment (nambrot) [inline CGPs/0020.md:8]: You'd probably wanna create a forum post and link it here

File: CGPs/0020.md (hunk 2 above)
Comment (anna-carroll) [reply CGPs/0020.md]: totally https://github.com/celo-org/celo-proposals/pull/155/commits/8dd23e7f0efa7ab00714a3e7087d98545a01eeae

File: CGPs/0020.md (hunk 1 above)
Comment (anna-carroll) [reply CGPs/0020.md]: good call 😄 https://github.com/celo-org/celo-proposals/pull/155/commits/027e9e46a1cd0b89e060f7a270932ab1f71166f2

File: CGPs/0020.md (hunk 3 above)
Comment (anna-carroll) [reply CGPs/0020.md]: cool, the link on the forum post will be broken until we merge this to master https://github.com/celo-org/celo-proposals/pull/155/commits/cdb1b17ac98e25b628014d5fbd34e6a1b7c5bdfb

File: CGPs/0021.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+## Proposed Changes
+
+1. Set Random randomnessBlockRetentionWindow to 60 * 60 * 24 / 5 (24 hours with 5 seconds per block)
+  - Destination: randomnessBlockRetentionWindow [setRandomnessBlockRetentionWindow](https://github.com/celo-org/celo-monorepo/blob/fb4f828a4e001b73a8c39aa34e3395f4015aaabc/packages/protocol/contracts/identity/Random.sol#L59)
+  - Data: 17280
+  - Value: 0
+
+## Verification
+
+- Call Random.sol `randomnessBlockRetentionWindow()` to see that the current parameter value is 720. After execution of this proposal, that value will be 24 times greater, 17280.
```
Comment (YazzyYaz) [inline CGPs/0021.md:27]: * Random -> `Random.sol`

File: CGPs/0021.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+- Status: DRAFT
+- Governance Proposal ID #: [if submitted]
+- Date Executed: [if executed]
+- Discussions To: https://forum.celo.org/t/discussions-for-cgp-21-update-randomnessblockretentionwindow-to-extend-attestation-expiration-duration/752
+
+## Overview
+
+This proposal will effectively extend the expiration of a requested attestation from 1 HOUR -> 1 DAY, improving upon [CGP 0015](https://github.com/celo-org/celo-proposals/blob/master/CGPs/0015.md) which previously attempted to do so.
+
+The initial [CGP 0015](https://github.com/celo-org/celo-proposals/blob/master/CGPs/0015.md) attempted to extend the expiration time of attestations from 1 hour to 24 hours by changing attestationExpiryBlocks in the Attestation contract. However, the Random contract only retains randomness for a period of blocks specified by randomnessBlockRetentionWindow. Though the Attestation contract was updated to be able to query randomness between 1 and 24 hours, the Random contract remains unable to serve those requests. Therefore, the expiration time for attestations remains functionally 1 hour.
```
Comment (YazzyYaz) [inline CGPs/0021.md:14]: Thank you for submitting your first CGP :) 

Here's the changes I recommend for this paragraph:
* attestationExpiryBlocks -> `attestationExpiryBlocks`
* randomnessBlockRetentionWindow -> `randomnessBlockRetentionWindow`
* Attestation -> `Attestations.sol`
* Random -> `Random.sol`

Comment (anna-carroll) [reply CGPs/0021.md]: https://github.com/celo-org/celo-proposals/pull/155/commits/b73cbc76624e95c338f0ca2fadf09b26517f08f7

File: CGPs/0021.md (hunk 4 above)
Comment (anna-carroll) [reply CGPs/0021.md]: https://github.com/celo-org/celo-proposals/pull/155/commits/b73cbc76624e95c338f0ca2fadf09b26517f08f7

**Review by YazzyYaz — APPROVED** [review]

**Pull request merged**